services/frontend-service/src/ui/Pages/Locks/LocksPage.test.tsx

```diff
@@ -40,120 +46,551 @@ describe('LocksPage', () => {
+                                locktest: {
+                                    message: 'locktest',
+                                    lockId: 'ui-v2-1337',
+                                    createdAt: new Date(1995, 11, 17),
+                                },
+                                lockbar: { message: 'lockbar', lockId: 'ui-v2-321', createdAt: new Date(1995, 11, 15) },
+                            },
+                        },
+                    },
+                },
+            },
+            filter: 'foo',
+            expectedLockIDs: ['ui-v2-123'],
+        },
+        {
+            name: 'get filtered locks (development, get 2 lock)',
+            envs: {
+                integration: {
+                    name: 'integration',
+                    locks: {},
+                    distanceToUpstream: 0,
+                    priority: 0,
+                    applications: {
+                        foo: {
+                            name: 'foo',
+                            version: 420,
+                            queuedVersion: 0,
+                            undeployVersion: false,
+                            locks: {
+                                lockfoo: { message: 'lockfoo', lockId: 'ui-v2-123', createdAt: new Date(1995, 11, 16) },
+                            },
+                        },
+                    },
+                },
+                development: {
+                    name: 'development',
+                    locks: {},
+                    distanceToUpstream: 0,
+                    priority: 0,
+                    applications: {
+                        bar: {
+                            name: 'bar',
+                            version: 1337,
+                            queuedVersion: 0,
+                            undeployVersion: false,
+                            locks: {
+                                locktest: {
+                                    message: 'locktest',
+                                    lockId: 'ui-v2-1337',
+                                    createdAt: new Date(1995, 11, 17),
+                                },
+                                lockbar: { message: 'lockbar', lockId: 'ui-v2-321', createdAt: new Date(1995, 11, 15) },
+                            },
+                        },
+                    },
+                },
+            },
+            filter: 'bar',
+            expectedLockIDs: ['ui-v2-321', 'ui-v2-1337'],
+        },
+    ];
+
+    describe.each(sampleAppFilteredData)(`Test Filtered Lock IDs`, (testcase) => {
+        it(testcase.name, () => {
+            // given
+            UpdateOverview.set({ environments: testcase.envs });
+            // when
+            const obtained = renderHook(() => useFilteredApplicationLockIDs(testcase.filter)).result.current;
+            // then
+            const sortedObtained = renderHook(() => sortAppLocksFromIDs(obtained, 'ascending')).result.current;
```

Comment:
Why do we implement a function `useFilteredApplicationLockIDs` that is only used in tests? Where is it use in the src?

Reply:
I'm still working on getting them to work in the src. I'm getting some hook specific errors